extension_templates/forecasting.py

```diff
@@ -238,19 +239,62 @@ def _update_predict_single(
 
     # todo: consider implementing this, optional
     # if not implementing, delete the method
-    def _compute_pred_int(self, alphas):
-        """Calculate the prediction errors for each point.
+    def _predict_interval(self, fh, X, coverage):
```

Comment:
right idea, although I think it may be confusing. In the extension template I'd only include `_predict_quantiles`, because `_predict_interval` automatically calls that. I'd add an explanation that the user can implement `_predict_interval` as well if there is a faster variant, but I think it is so rare that it shouldn't be in the template as a "fill in" function, but only mentioned in the body of `_predict_quantiles`.